src/java/org/apache/cassandra/cql3/statements/BatchStatement.java

```diff
@@ -357,7 +357,7 @@ public List<? extends IMutation> getMutations(ClientState state,
         }
         // local is either executeWithoutConditions modifying a virtual table (doesn't support txns) or executeLocal
         // which is called by test or internal things that are bypassing distributed system modification/checks
-        return collector.toMutations(state, local ? PotentialTxnConflicts.ALLOW : PotentialTxnConflicts.DISALLOW);
+        return collector.toMutations(state, local ? PotentialTxnConflicts.ALLOW : PotentialTxnConflicts.DISALLOW, attrs.isTimestampSet());
```

Comment:
this is dead code atm, i pass it in but its not used.

```
BEGIN BATCH
   INSERT INTO tbl(pk, ck, v) VALUES (0, 0, 0) USING TIMESTAMP 42;
   INSERT INTO tbl(pk, ck, v) VALUES (0, 1, 1);
```

in this example the first mutation is using the user timestamp, but the second is using the server timestamp; they are the same `Mutation` object as they share the same PK!